FBSnapshotTestCase/FBSnapshotTestCase.h

```diff
@@ -22,46 +24,66 @@
  Similar to our much-loved XCTAssert() macros. Use this to perform your test. No need to write an explanation, though.
  @param view The view to snapshot
  @param identifier An optional identifier, used if there are multiple snapshot tests in a given -test method.
- @param referenceImageDirectorySuffix An optional suffix, appended to the reference image directory path, such as "_iOS8"
+ @param version An optional suffix, appended to the reference image directory path, such as "_iOS8"
+ @param architectureTypes An optional dictionary with a type and suffix e.g. @{ FBSnapshotTestArchitectureType.bit64 : @"_64"}
  */
-#define FBSnapshotVerifyViewWithReferenceDirectorySuffix(view__, identifier__, referenceImagesDirectorySuffix__) \
+#define FBSnapshotVerifyViewWithOptions(view__, identifier__, version__, architectureTypes__) \
 { \
+NSOrderedSet *suffixes__ = FBSuffixesForArchitectureTypes(architectureTypes__); \
 NSError *error__ = nil; \
-NSString *referenceImagesDirectory__ = [NSString stringWithFormat:@"%s%@", FB_REFERENCE_IMAGE_DIR, referenceImagesDirectorySuffix__]; \
-BOOL comparisonSuccess__ = [self compareSnapshotOfView:(view__) referenceImagesDirectory:referenceImagesDirectory__ identifier:(identifier__) error:&error__]; \
+BOOL comparisonSuccess__; \
+for (NSString *suffix__ in suffixes__) { \
+NSString *referenceImagesDirectory__ = [NSString stringWithFormat:@"%s%@%@", FB_REFERENCE_IMAGE_DIR, version__, suffix__]; \
+comparisonSuccess__ = [self compareSnapshotOfView:(view__) referenceImagesDirectory:referenceImagesDirectory__ identifier:(identifier__) error:&error__]; \
+if (comparisonSuccess__) break; \
+} \
 XCTAssertTrue(comparisonSuccess__, @"Snapshot comparison failed: %@", error__); \
-XCTAssertFalse(self.recordMode, @"Test ran in record mode. Reference image is now saved. Disable record mode to perform an actual snapshot comparsion!"); \
+XCTAssertFalse(self.recordMode, @"Test ran in record mode. Reference image is now saved. Disable record mode to perform an actual snapshot comparison!"); \
 }
 
 #define FBSnapshotVerifyView(view__, identifier__) \
 { \
-FBSnapshotVerifyViewWithReferenceDirectorySuffix(view__, identifier__, @""); \
+NSDictionary *architectureTypes__ = @{ \
+FBSnapshotTestArchitectureType32BitKey : @"_32", \
+FBSnapshotTestArchitectureType64BitKey : @"_64", \
+}; \
```

Comment:
For now I went with simply `NSString`s in a dictionary:

key : suffix

I tried `NS_Enum` but didn't feel good.